python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -310,6 +314,11 @@ def select_operator_to_run(
         if state.num_queued() > 0 and _execution_allowed(op, cur_usage, limits)
     ]
 
+    # If no ops are allowed to execute due to resource constraints, try to trigger
+    # cluster scale-up.
+    if not ops:
+        _try_to_scale_up_cluster(topology)
```

Comment:
This could be inaccurate if num_queued() == 0 for all ops. For example, if there was just a single Map stage, the queue could be zero, but we don't need to scale up, since we are just waiting for existing tasks to finish.

Reply:
Hmm I thought that I had accounted for that with the way that `_try_to_scale_up_cluster()` is designed, where it only requests incremental resource for ops with `num_queued > 0`? This should bottom-out at making cluster resource requests with the current resource usage, which should have a 1:1 correspondence with already-allocated resources in the cluster and should therefore be a no-op, right?

The one exception would possibly be object store memory, since that doesn't have a 1:1 correspondence with actual usage.

I could always change this check to `if not ops and any(state.num_queued() > 0 for state in topology.values())` to make this explicit.

Reply:
The rounding up with `ceil` may make it exceed already allocated now. Added additional condition of checking at least one input queue non-empty.